### PR TITLE
DEV: Introduce post_with_secure_uploads? plugin modifier

### PR DESCRIPTION
**Pull request by martin-brennan**
This modifier allows plugins to alter the outcome of
`with_secure_uploads?` on a Post record, for cases when
plugins need post-attached uploads to always be secure (or
not secure) in specific scenarios.

Also makes a slight change to `TopicUploadSecurityManager`
to return rebaked posts.

### CODE REVIEW

**Review by SamSaffron — APPROVED** [review]

**Comment by SamSaffron** [comment]
looks good but one general concern I have is the name `with_secure_uploads?` feels somewhat confusing. 

Maybe `has_secured_uploads?` it is just you feel like it should be returning some sort of set of things with the `with` word.

File: lib/topic_upload_security_manager.rb [hunk 1]
```diff
@@ -35,14 +36,18 @@ def run
...
+          rebaked_posts << post
+        end
         Rails.logger.debug(
           "Security updated & rebake complete in topic #{@topic.id} - post #{post.id}",
         )
       end
     end
 
-    return if !SiteSetting.secure_uploads
+    return rebaked_posts if !SiteSetting.secure_uploads
```
Comment (Drenmi) [inline lib/topic_upload_security_manager.rb:50]: 👍

File: app/models/post.rb [hunk 2]
```diff
@@ -565,11 +565,26 @@ def reviewable_flag
...
+    # is not an option. This also is not taken into account in the secure upload
+    # rake tasks, and will more than likely change in future.
+    modifier_result =
+      DiscoursePluginRegistry.apply_modifier(
+        :post_with_secure_uploads?,
+        nil,
+        self,
+        topic_including_deleted,
+      )
+    return modifier_result if modifier_result != nil
```
Comment (Drenmi) [inline app/models/post.rb:586]: I usually do:

```suggestion
    return modifier_result if !modifier_result.nil?
```

But it's really a tomato-tomato thing. 🙂

**Review by Drenmi — APPROVED** [review]
Looks useful! 👍

**Comment by martin-brennan** [comment]
@SamSaffron you make a good point about `with_secure_uploads?` -- it was made that way in the original PR https://github.com/discourse/discourse/commit/102909edb3376d41398a8ca89d821b2e670bcecc  and I never really thought about it. `has_secure_uploads?` works better for sure -- I will make another PR to rename it to `has_secure_uploads?` before this one and ping you there.

Thanks for the review too @Drenmi I will fixup :+1:

**Pull request merged**